### PR TITLE
Additional CosmosDBInputAttribute constructors

### LINKED ISSUE #1410: [SDK Binding - Cosmos] Improve `CosmosDBInput` attribute requirements
1. Input binding attribute should not require `databaseName` and `containerName` for `CosmosClient` target type

We do not need the database name or container name  to create a `CosmosClient`:

```csharp
[Function(nameof(DocsByUsingCosmosClient))]
public async Task<HttpResponseData>  DocsByUsingCosmosClient(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequestData req,
        [CosmosDBInput(Connection = "CosmosDBConnection")] CosmosClient client)
```

2. Input binding attribute should not require `containerName` for `Database` target type client

We do not need the container name to create a `Database` client:

```csharp
[Function(nameof(DocsByUsingDatabaseClient))]
public async Task<HttpResponseData> DocsByUsingDatabaseClient(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequestData req,
        [CosmosDBInput("ToDoItems", Connection = "CosmosDBConnection")] Database database)
```

### PR DESCRIPTION
**Pull request by liliankasem**
### Issue describing the changes in this PR

resolves #1410

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Was worried about making DatabaseName and ContainerName nullable so ended up with this - open to better suggestions

### CODE REVIEW

File: extensions/Worker.Extensions.CosmosDB/src/CosmosDBInputAttribute.cs [hunk 1]
```diff
@@ -11,11 +12,34 @@ namespace Microsoft.Azure.Functions.Worker
...
     public sealed class CosmosDBInputAttribute : InputBindingAttribute
     {
-        /// <summary>
+        /// <Summary>
         /// Constructs a new instance.
+        /// Use this constructor when binding to a CosmosClient.
+        /// </summary>
+        public CosmosDBInputAttribute()
+        {
+            DatabaseName = string.Empty;
```
Comment (surgupta-msft) [inline extensions/Worker.Extensions.CosmosDB/src/CosmosDBInputAttribute.cs:21]: Can an alternative approach be to assign `DefaultValue` attribute as below? https://github.com/Azure/azure-functions-dotnet-worker/blob/9efb07783a1b8769fd87ef4199d356544bf0bf2d/extensions/Worker.Extensions.Storage.Blobs/src/BlobInputAttribute.cs#L32-L33

File: extensions/Worker.Extensions.CosmosDB/src/CosmosDBInputAttribute.cs [hunk 2]
```diff
@@ -11,11 +12,34 @@ namespace Microsoft.Azure.Functions.Worker
     [InputConverter(typeof(CosmosDBConverter))]
     public sealed class CosmosDBInputAttribute : InputBindingAttribute
     {
-        /// <summary>
+        /// <Summary>
```
[the remark this answers is not in the document]
Comment (liliankasem) [reply extensions/Worker.Extensions.CosmosDB/src/CosmosDBInputAttribute.cs:15]: accident! I'll change to lowercase

File: extensions/Worker.Extensions.CosmosDB/src/CosmosDBInputAttribute.cs (hunk 1 above)
Comment (liliankasem) [reply extensions/Worker.Extensions.CosmosDB/src/CosmosDBInputAttribute.cs:21]: Let me try it out :)

Comment (liliankasem) [reply extensions/Worker.Extensions.CosmosDB/src/CosmosDBInputAttribute.cs:21]: Doesn't quiet do the job:

`There is no argument given that corresponds to the required parameter 'databaseName' of 'CosmosDBInputAttribute.CosmosDBInputAttribute(string, string)' [/Users/likasem/source/functions/azure-functions-dotnet-worker/samples/WorkerBindingSamples/WorkerBindingSamples.csproj`

**Review by jviau — APPROVED** [review]

**Pull request merged**